### PR TITLE
Move upb generated code into separate libraries

### LINKED ISSUE #21639: The Node build breaks with the latest core
When I tried to update the Node library to use the current version of the core in grpc/grpc-node#1232, the build fails with the following error:

```
gyp: Duplicate basenames in sources section, see list above
static library binding.gyp:grpc#target has several files with the same basename:
  http.upb: deps/grpc/src/core/ext/upb-generated/google/api/http.upb.c deps/grpc/src/core/ext/upb-generated/envoy/type/http.upb.c
libtool on Mac cannot handle that. Use --no-duplicate-basename-check to disable this validation.
```

I tried the suggested argument and it didn't help, and I expect that if it did, the Mac build would still have problems.

### PR DESCRIPTION
**Pull request by murgatroid99**
As described in #21702, the Node build fails when two different files in the same static library have the same base name. Currently, this is happening in one instance because the `envoy_type_upb` and `google_api_upb` filegroups both have a `http.upb.c`. Moving them into separate library targets solves that problem. For consistency, this change moves all upb generated file sets into separate library targets.

This fixes #21639.

### CODE REVIEW

File: BUILD.gn [hunk 1]
```diff
@@ -909,7 +833,12 @@ config("grpc_config") {
         "//third_party/boringssl",
         "//third_party/zlib",
         ":gpr",
+        ":alts_upb",
         ":upb",
+        ":grpc_health_upb",
```
Comment (nicolasnoble) [inline BUILD.gn]: It looks like these aren't being created into this file.

Then again, we provided it for courtesy, and aren't testing it nor actively maintaining it.

Comment (murgatroid99) [reply BUILD.gn]: I don't understand how that file is generated or what it's supposed to contain. But if this change breaks it, that implies that it is determining the libraries it needs separately from the dependency tree defined in `build.yaml`, which is an unsustainably fragile code generation strategy.

**Comment by nicolasnoble** [comment]
> This looks fine to me, but I'll let Nico have the final say, since he knows the build system so much better than I do.

This PR's a result of a conversation between him and me anyway :)

**Comment by nicolasnoble** [comment]
cc @veblush

**Comment by murgatroid99** [comment]
With the latest commits this now makes more significant changes to the Makefile template. Specifically, it removes all special handling for abseil. Instead, transitive dependencies are linked when building all targets.

**Comment by veblush** [comment]
gRPC artifacts from makefile are supposed to embed all necessary object files such as `gpr`, `zlib`, `upb`, and so on so that it can be used as a standalone library.

```
  % if lib.get('baselib', False):
   $(LIBGPR_OBJS) \
   $(LIBGRPC_ABSEIL_OBJS) \
   $(ZLIB_MERGE_OBJS) \
   $(CARES_MERGE_OBJS) \
   $(ADDRESS_SORTING_MERGE_OBJS) \
   $(UPB_MERGE_OBJS) \
  % if lib.get('secure', 'check') == True:
   $(OPENSSL_MERGE_OBJS) \
  % endif
```

But this PR changes this behavior and [tests](https://source.cloud.google.com/results/invocations/580cf62a-7902-47ea-ad89-d32871d419aa/targets/github%2Fgrpc/tests;group=tasks;test=distribtest.cpp_linux_x64_jessie_routeguide;row=2) are not failing because of this.

**Comment by jtattermusch** [comment]
this will probably need a revamp after removal of build.yaml.
https://github.com/grpc/grpc/pull/21929

**Pull request closed without merging**